Plugins/Public/mobiledocking_plugin/Main.cpp

```diff
@@ -192,266 +89,700 @@ void LoadSettings()
 			{
 				while (ini.read_value())
 				{
-					if (ini.is_value("allowedmodule"))
+					if (ini.is_value("module"))
 					{
-						dockingModuleEquipmentIds.push_back(CreateID(ini.get_value_string()));
+						dockingModuleEquipmentCapacityMap[CreateID(ini.get_value_string(0))] = ini.get_value_int(1);
 						dockingModAmount++;
 					}
 					else if (ini.is_value("cargo_capacity_limit"))
 					{
 						cargoCapacityLimit = ini.get_value_int(0);
 					}
+					else if (ini.is_value("carrier_data_wipe_period"))
+					{
+						forgetCarrierDataInSeconds = ini.get_value_int(0);
+					}
+					else if (ini.is_value("mobile_dock_range"))
+					{
+						mobileDockingRange = ini.get_value_float(0);
+					}
+					else if (ini.is_value("docking_period"))
+					{
+						dockingPeriod = ini.get_value_int(0);
+					}
+					else if (ini.is_value("docking_tolerance"))
+					{
+						maxDockingDistanceTolerance = ini.get_value_float(0);
+					}
+					else if (ini.is_value("default_return_system"))
+					{
+						defaultReturnSystem = ini.get_value_string(0);
+					}
+					else if (ini.is_value("default_return_base"))
+					{
+						defaultReturnBase = ini.get_value_string(0);
+					}
+					else if (ini.is_value("target_mobile_docking_base"))
+					{
+						mobileDockingProxyBase = CreateID(ini.get_value_string(0));
+					}
+					else if (ini.is_value("disable_docking_repairs"))
+					{
+						disableRegensOnDockAttempt = ini.get_value_bool(0);
+					}
+					else if (ini.is_value("docking_shield_drain"))
+					{
+						disableShieldsOnDockAttempt = ini.get_value_bool(0);
+					}
 				}
 			}
 		}
 		ini.close();
 	}
 
+	if (ini.open(scCarrierDataFile.c_str(), false))
+	{
+		std::lock_guard<std::mutex> saveLock(saveMutex);
+		time_t curTime = time(0);
+		while (ini.read_header())
+		{
+			if (ini.is_header("CarrierData"))
+			{
+				CARRIERINFO ci;
+				wstring carrierName;
+				boolean doLoad = true;
+				while (ini.read_value() && doLoad)

```

Comment:
don't we want to do this while ini.read_value() regardless of doLoad? Since we'll want a complete list of docked ships for the MoveOfflineShipToLastDockedSolar caller below

Reply:
Excellent point, adjusting.